core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala

```diff
@@ -268,6 +282,29 @@ class TransactionMarkerChannelManager(config: KafkaConfig,
     addTxnMarkersToBrokerQueue(transactionalId, txnMetadata.producerId, txnMetadata.producerEpoch, txnResult, coordinatorEpoch, txnMetadata.topicPartitions.toSet)
   }
 
+  private def tryAppendToLog(txnLogAppend: TxnLogAppend) = {
+    // try to append to the transaction log
+    def retryAppendCallback(error: Errors): Unit =
```

Comment:
This probably should be named `appendCallback` since it would be invoked for all attempts, including those which are not retries.